### DISCUSSION TITLE
Date Due date value becomes strange in recurring transactions

### DISCUSSION THREAD

**Discussion by DevAnyKR** [moneymanagerex/moneymanagerex#5889] [Q&A]
for use my Favorite Transaction feature

![스크린샷 2023-05-05 232406](https://user-images.githubusercontent.com/110871727/236485422-10d1f46d-be5d-47b9-84ab-7cf4564e6bee.png)

![스크린샷 2023-05-05 232526](https://user-images.githubusercontent.com/110871727/236485790-ef345b06-0afd-4488-9b02-6aa0d4fa3202.png)

1. set 9999-12-31
2. Enter next Occurrence...
3. 1000--0--3

😮

How do I keep it at 9999-12-31?
Because date due shouldn't change

**Comment by tactilis** [comment]
The Date Due is changing because it is incrementing to the _next_ Due occurrence after you have done an `Enter next Occurrence...`
Because you have chosen an initial Date Due of 9999-12-31, the field overflows when it increments.

I would set the initial Date Due to something like 2099-01-01, the `Repeat` to Daily. The Date Due will then increment by just one day each time you enter an occurrence.

**Reply by DevAnyKR** [reply]
+1m becomes 10000-01-31
I expected it to be displayed as 0000-01-31.
I don't know why the first digit is 1.
similar to shift right "date string >> 1 & ####-##-##"

**Reply by Macaulay1234** [reply]
> +1m becomes 10000-01-31
> I expected it to be displayed as 0000-01-31.
> I don't know why the first digit is 1.
> similar to shift right "date string >> 1 & ####-##-##"

**Comment by vomikan** [comment]
let's return to this problem in 7,976 years.

**Comment by Macaulay1234** [comment]
Yes

**Comment by Macaulay1234** [comment]
> The Date Due is changing because it is incrementing to the _next_ Due occurrence after you have done an `Enter next Occurrence...`
> Because you have chosen an initial Date Due of 9999-12-31, the field overflows when it increments.
> 
> I would set the initial Date Due to something like 2099-01-01, the `Repeat` to Daily. The Date Due will then increment by just one day each time you enter an occurrence.
> 
>

**Reply by DevAnyKR** [reply]
I already know that, but it's not normal either, it's just a trick. Don't need a due date when I use like favorites transaction. 
The content of my question is why an abnormal date appears.
